# Primer_Blast_like

**Aim:**

To carry out *NCBI Primer-Blast like* function locally and with not-yet published genome database, and maybe for further use in the future.

**Required environment:**

1. Python3

2. NCBI blast+

3. Database created with NCBI+

4. Certain directory in order for outputing

5. Python Packages: 

    * Biopython

    * os

    * pandas

    * primer3

**Required parameters:**

1. email (in case of contact from NCBI)

2. gene id, database (to download sequence from NCBI genbank)

>[NCBI efetch database](https://www.ncbi.nlm.nih.gov/books/NBK25497/table/chapter2.T._entrez_unique_identifiers_ui/?report=objectonly)

3. output address, database address (to save result)

4. evalue, identity (to check Specificity)


**Optimal parameters: (for designing primers)**

1. gene name (to name saved files)

2. primer3 parameters (for special need in designing primers)
